api/controller.py

```diff
@@ -828,14 +830,40 @@ def _load_search_facets(self, lane):
             default_entrypoint=default_entrypoint,
         )
 
-    def search(self, lane_identifier):
-
+    def search(self, lane_identifier, feed_class=AcquisitionFeed):
+        """Search for books."""
         lane = self.load_lane(lane_identifier)
         if isinstance(lane, ProblemDetail):
             return lane
+
+        pagination = load_pagination_from_request(
+            default_size=Pagination.DEFAULT_SEARCH_SIZE
+        )
+        if isinstance(pagination, ProblemDetail):
+            return pagination
+
+        facets = self._load_search_facets(lane)
+        if isinstance(facets, ProblemDetail):
+            return lane
+
+        # TODO: Is this still necessary? Most other feeds now come
+        # from the search index, and those methods let lower-down code
+        # find or create the ExternalSearchIndex object.
+        #
+        # The argument against is that it's inefficient to look up
+        # the ExternalSearchIndex object every time. I think this is only
+        # valid if creating the object requires a back-and-forth with the
+        # search server.
```

Comment:
Yeah, I think this was because the initial setup was slow. It is still necessary to handle the case where the search engine is misconfigured, but maybe not like this.